qt/widgets/common/src/FitPropertyBrowser.cpp

```diff
@@ -3526,5 +3526,14 @@ QStringList FitPropertyBrowser::getPeakPrefixes() const {
   return peaks;
 }
 
+/*
+ Clears the editors managers and factories.
+ This is not done in the destructor as properties can still be used elsewhere.
+ This is only for use in testing.
+ */
+void FitPropertyBrowser::clearEditorManagersAndFactories() {
```

Comment:
Could you help me understand why we can't do something in the destructor? I can see that a static map of the managers is maintained so it needs to be cleared. My concern here is that it has to be done manually. Do we not need to do something similar in workbench if we are constructing multiple browsers in each plot? I think in MantidPlot we got away with it by only ever constructing a single instance.

If I understood correctly is it the case that we need to make sure the [`setFactoryForManager`](https://github.com/mantidproject/mantid/blob/master/qt/widgets/common/src/FitPropertyBrowser.cpp#L604) registrations in [`createEditors`](https://github.com/mantidproject/mantid/blob/master/qt/widgets/common/src/FitPropertyBrowser.cpp#L592) are only done once?

Reply:
So my thinking was that we can't just clear it in the destructor because in workbench we can have multiple instances which are all stored in the same map so clearing it would remove the managers and factories for any plots which remain open. However it should now work if I unset each of the factories for the managers in the destructor (I had previously tried this but that was before I had found the repeated init so it was still causing a problem).  I'll change it to do this instead.